web/libs/ui/src/lib/select/select.tsx

```diff
@@ -281,72 +352,22 @@ export const Select = forwardRef(
-                          {multiple ? (
-                            <Option
-                              multiple={multiple}
-                              label={label}
-                              isIndeterminate={!isOptionSelected && isIndeterminate}
-                              isOptionSelected={isOptionSelected}
-                              onSelect={() => {
-                                children.forEach((child: SelectOption<T>) => {
-                                  const childVal = child?.value ?? child;
-                                  isOptionSelected ? _onChange(childVal, true) : _onChange(childVal, false);
-                                });
-                              }}
-                            />
-                          ) : (
-                            <div className="pl-3 font-bold text-neutral-content-subtler pt-2">{label}</div>
-                          )}
-                          <div className="pl-2">
-                            {children.map((item, i) => {
-                              const val = item?.value ?? item;
-                              const lab = item?.label ?? val;
-                              const isChildOptionSelected = isSelected(val);
-                              return (
-                                <Option
-                                  key={`${val}_${i}`}
-                                  value={val}
-                                  label={lab}
-                                  isOptionSelected={isChildOptionSelected}
-                                  disabled={item?.disabled}
-                                  style={item?.style}
-                                  multiple={multiple}
-                                  onSelect={() => {
-                                    _onChange(val, isChildOptionSelected);
-                                  }}
-                                />
-                              );
-                            })}
-                          </div>
-                        </CommandGroup>
-                      );
-                    }
-                    return (
-                      <Option
-                        key={`${optionValue}_${index}`}
-                        value={optionValue}
-                        label={label}
-                        isOptionSelected={isOptionSelected}
-                        disabled={option?.disabled}
-                        style={option?.style}
-                        multiple={multiple}
-                        onSelect={() => {
-                          _onChange(optionValue, isOptionSelected);
-                        }}
-                      />
-                    );
-                  })}
+                  {isVirtualList ? (
+                    <VariableSizeList
+                      itemData={renderedOptions}
+                      itemSize={() => 40}
+                      itemCount={renderedOptions.length}
+                      height={200}
```

Comment:
Can this be adapted to being a non hard-coded height? I'm thinking that is what is causing the list sizes smaller than this height to have empty space, particularly on searching but I would imagine it would happen anytime the options would be only a few.

Reply:
yeah i hard coded it for now but I'm wondering if we should have it as a prop and put the responsibility on the wrapper component to define how high that block should be - this way if we want to dynamically update - thats fine - if we want to set it to something that makes sense for that component - thats fine too